chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -60,34 +71,137 @@ The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/b
+5. Restart your Lotus node.
+6. Backfill the `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices using the `lotus-shed index backfill-*` CLI tooling available in the [previous indexing system](#previous-indexing-system) for the range of epochs between the upgrade to `ChainIndexer` and the rollback of `ChainIndexer`.
+7. Route traffic back to the node.  
+
+## Terminology
+### Previous Indexing System
+* This corresponds to the indexing system used in Lotus versions before v1.31.0.  
+* It has been replaced by the [ChainIndexer](#chainindexer-indexing-system).
+* It was composed of three indexers using three separate databases: [`EthTxHashLookup`](https://github.com/filecoin-project/lotus/blob/v1.31.0/chain/ethhashlookup/eth_transaction_hash_lookup.go), [`MsgIndex`](https://github.com/filecoin-project/lotus/blob/v1.31.0/chain/index/msgindex.go), and [`EventIndex`](https://github.com/filecoin-project/lotus/blob/v1.31.0/chain/events/filter/index.go).
+* It persisted state to the [removed option](#removed-options) for `Events.DatabasePath`, which defaulted to `${LOTUS_PATH}/sqlite`.
+* It had CLI backfill tooling: `lotus-shed index backfill-*`
+
+### ChainIndexer Indexing System
+* This corresponds to the indexing system used in Lotus versions v1.31.0 onwards.  
+* It replaced the [previous indexing system](#previous-indexing-system).
+* It is composed of a single indexer, [`ChainIndexer`](https://github.com/filecoin-project/lotus/blob/master/chain/index/indexer.go), using a [single database for transactions, messages, and events](https://github.com/filecoin-project/lotus/blob/master/chain/index/ddls.go).
+* It persists state to `${LOTUS_HOME}/chainindex`.
+* It has this CLI backfill tooling: [`lotus-shed chainindex validate-backfill`](#lotus-shed-chainindex-validate-backfill-cli-tool)
+* **Storage requirements:** See the [backfill disk space requirements](#backfill-disk-space-requirements).
+* **Backfil times:** See the [backfill timing](#backfill-timing).
+
+## Appendix
+
+### Wny isn't there an automated migration from the [previous indexing system](#previous-indexing-system) to the [ChainIndexer indexing system](#chainindexer-indexing-system)?
+
+The decision not invest here ultimately comes down to the development-time cost vs. benefit ratio.
+
+For achival nodes, we don't have the confidence that the [previous indexing system](#previous-indexing-system) has the correct data to bootstrap from. In 2024, Lotus maintainers have fixed multiple bugs in the [previous indexing system](#previous-indexing-system), but they still see reports of smissing data, mismatched event index counts, etc.  Investing here in a migration isn't guaranteed to yield a correct index. As a result, one would still need to perform the [backfill steps](#backfill).  While this should be faster having partially correct data than no data, it would still require an archival node to take an outage on the order of days which isn't good enough.
```

Comment:
```suggestion
For achival nodes, we don't have the confidence that the [previous indexing system](#previous-indexing-system) has the correct data to bootstrap from. In 2024, Lotus maintainers have fixed multiple bugs in the [previous indexing system](#previous-indexing-system), but they still see reports of missing data, mismatched event index counts, etc.  Investing here in a migration isn't guaranteed to yield a correct index. As a result, one would still need to perform the [backfill steps](#backfill).  While this should be faster having partially correct data than no data, it would still require an archival node to take an outage on the order of days which isn't good enough.
```